File: server/src/participant/participant.controller.ts
import { Body, Controller, Param, Post, Put } from '@nestjs/common';

import { CreateParticipantUseCase } from './useCases/CreateParticipantUseCase';
import { CreateParticipantDTO } from './dtos/CreateParticipantDTO';
import { UpdateParticipantUseCase } from './useCases/UpdateParticipantUseCase';
import { UpdateParticipantDTO } from './dtos/UpdateParticipantDTO';

@Controller('/participants')
export class ParticipantController {
  constructor(
    private readonly createParticipantUseCase: CreateParticipantUseCase,
    private readonly updateParticipantUseCase: UpdateParticipantUseCase,
  ) {}

  @Post('/')
  async create(@Body() createParticipantDTO: CreateParticipantDTO) {
    const participant =
      await this.createParticipantUseCase.execute(createParticipantDTO);

    return participant;
  }

  @Put('/:participantId')
  async updateById(
    @Param('participantId') participantId: string,
    @Body() updateParticipantDTO: UpdateParticipantDTO,
  ) {
    const participantUpdated = await this.updateParticipantUseCase.execute(
      participantId,
      updateParticipantDTO,
    );

    return participantUpdated;
  }
}

File: microservices/participants-service/src/main/java/com/sportsfinance/participants/ParticipantController.java
package com.sportsfinance.participants;

import com.sportsfinance.participants.dtos.CreateParticipantDTO;
import com.sportsfinance.participants.dtos.RecoveryParticipantsDTO;
import com.sportsfinance.participants.dtos.UpdateParticipantDTO;
import com.sportsfinance.participants.mappers.ParticipantMapper;
import com.sportsfinance.participants.services.CreateParticipantService;
import com.sportsfinance.participants.services.UpdateParticipantService;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/participants")
public class ParticipantController {

    @Autowired
    private CreateParticipantService createParticipantService;

    @Autowired
    private UpdateParticipantService updateParticipantService;

    @Autowired
    ParticipantMapper participantMapper;

    @PostMapping
    public ResponseEntity<RecoveryParticipantsDTO> create(@RequestBody() @Valid CreateParticipantDTO createParticipantDTO) {
        RecoveryParticipantsDTO participant = this.createParticipantService.execute(createParticipantDTO);

        return new ResponseEntity<>(participant, HttpStatus.CREATED);
    }

    @PutMapping("/{participantId}")
    public ResponseEntity<RecoveryParticipantsDTO> updateById(
            @PathVariable String participantId, @RequestBody() @Valid UpdateParticipantDTO updateParticipantDTO
    ) {
        RecoveryParticipantsDTO participantUpdated = this.updateParticipantService.execute(
                participantId,
                updateParticipantDTO
        );

        return new ResponseEntity<>(participantUpdated, HttpStatus.OK);
    }

}
